src/circuits/mimc/round.hpp

```diff
@@ -0,0 +1,47 @@
+// DISCLAIMER:
+// Content taken and adapted from:
+// https://github.com/HarryR/ethsnarks/blob/master/src/gadgets/mimc.hpp
+
+#ifndef __ZETH_MIMC_ROUND_HPP__
+#define __ZETH_MIMC_ROUND_HPP__
+
+#include "snarks_alias.hpp"
+#include "circuits/circuits-util.hpp"
+
+namespace libzeth {
+ /*
+  * MiMCe7_round_gadget enforces correct computation of a MiMC permutation round with exponent 7.
+  * In MiMC permutation last round differs from the others since the key is added again. We use a boolean variable `add_k_to_result` to manage this case.
+  */
+template<typename FieldT>
+class MiMCe7_round_gadget : public libsnark::gadget<FieldT> {
+public:
+    const libsnark::pb_variable<FieldT> x;  // round message
+    const libsnark::pb_variable<FieldT> k;  // round key
+    const FieldT C; // round constant
+    const bool add_k_to_result; // variable to add the key after the round
+    libsnark::pb_variable<FieldT> a;  // constraint t^2 variable
+    libsnark::pb_variable<FieldT> b;  // constraint t^4 variable
+    libsnark::pb_variable<FieldT> c;  // constraint t^6 variable
+    libsnark::pb_variable<FieldT> d;  // constraint t^7 variable
```

Comment:
`b` -> `t4`?
`c` -> `t6`?
`d` -> `t7`?

Reply:
Just added "intermediary variable" heading comment and removed the ones on the side since in the `.tcc` is explained how the variable are used.